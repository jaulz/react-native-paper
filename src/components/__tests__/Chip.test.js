/* @flow */

import * as React from 'react';
import renderer from 'react-test-renderer';
import Chip from '../Chip';

it('renders chip with onPress', () => {
  const tree = renderer
    .create(<Chip onPress={() => {}}>Example Chip</Chip>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders chip with icon', () => {
  const tree = renderer.create(<Chip icon="info">Example Chip</Chip>).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders deletable chip', () => {
  const tree = renderer
    .create(
      <Chip icon="info" onDelete={() => {}}>
        Example Chip
      </Chip>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders outlined disabled chip', () => {
  const tree = renderer
    .create(
      <Chip mode="outlined" disabled>
        Example Chip
      </Chip>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders selected chip', () => {
  const tree = renderer.create(<Chip selected>Example Chip</Chip>).toJSON();

  expect(tree).toMatchSnapshot();
});
