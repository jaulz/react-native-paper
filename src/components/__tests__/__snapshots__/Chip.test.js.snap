// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders chip with icon 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ebebeb",
      "borderColor": "#ebebeb",
      "borderRadius": 16,
      "borderStyle": "solid",
      "borderWidth": 0.5,
      "elevation": 0,
    }
  }
>
  <View
    accessibilityComponentType="button"
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderRadius": 16,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "paddingHorizontal": 4,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "padding": 4,
            },
            null,
          ]
        }
      >
        <Text
          accessibilityElementsHidden={true}
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          style={
            Array [
              Object {
                "color": "rgba(0, 0, 0, 0.54)",
                "fontSize": 18,
              },
              Array [
                Object {
                  "transform": Array [
                    Object {
                      "scaleX": 1,
                    },
                  ],
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "Helvetica Neue",
              "textAlign": "left",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "height": 24,
                "lineHeight": 24,
                "marginVertical": 4,
                "textAlignVertical": "center",
              },
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
                "marginLeft": 4,
                "marginRight": 8,
              },
            ],
          ]
        }
        theme={
          Object {
            "colors": Object {
              "accent": "#03dac4",
              "backdrop": "rgba(0, 0, 0, 0.5)",
              "background": "#f6f6f6",
              "disabled": "rgba(0, 0, 0, 0.26)",
              "error": "#B00020",
              "placeholder": "rgba(0, 0, 0, 0.54)",
              "primary": "#6200ee",
              "surface": "#ffffff",
              "text": "#000000",
            },
            "dark": false,
            "fonts": Object {
              "light": "HelveticaNeue-Light",
              "medium": "HelveticaNeue-Medium",
              "regular": "Helvetica Neue",
              "thin": "HelveticaNeue-Thin",
            },
            "roundness": 4,
          }
        }
      >
        Example Chip
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders chip with onPress 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ebebeb",
      "borderColor": "#ebebeb",
      "borderRadius": 16,
      "borderStyle": "solid",
      "borderWidth": 0.5,
      "elevation": 0,
    }
  }
>
  <View
    accessibilityComponentType="button"
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderRadius": 16,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "paddingHorizontal": 4,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "Helvetica Neue",
              "textAlign": "left",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "height": 24,
                "lineHeight": 24,
                "marginVertical": 4,
                "textAlignVertical": "center",
              },
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
                "marginLeft": 8,
                "marginRight": 8,
              },
            ],
          ]
        }
        theme={
          Object {
            "colors": Object {
              "accent": "#03dac4",
              "backdrop": "rgba(0, 0, 0, 0.5)",
              "background": "#f6f6f6",
              "disabled": "rgba(0, 0, 0, 0.26)",
              "error": "#B00020",
              "placeholder": "rgba(0, 0, 0, 0.54)",
              "primary": "#6200ee",
              "surface": "#ffffff",
              "text": "#000000",
            },
            "dark": false,
            "fonts": Object {
              "light": "HelveticaNeue-Light",
              "medium": "HelveticaNeue-Medium",
              "regular": "Helvetica Neue",
              "thin": "HelveticaNeue-Thin",
            },
            "roundness": 4,
          }
        }
      >
        Example Chip
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders deletable chip 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ebebeb",
      "borderColor": "#ebebeb",
      "borderRadius": 16,
      "borderStyle": "solid",
      "borderWidth": 0.5,
      "elevation": 0,
    }
  }
>
  <View
    accessibilityComponentType="button"
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderRadius": 16,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "paddingHorizontal": 4,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "padding": 4,
            },
            null,
          ]
        }
      >
        <Text
          accessibilityElementsHidden={true}
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          style={
            Array [
              Object {
                "color": "rgba(0, 0, 0, 0.54)",
                "fontSize": 18,
              },
              Array [
                Object {
                  "transform": Array [
                    Object {
                      "scaleX": 1,
                    },
                  ],
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "Helvetica Neue",
              "textAlign": "left",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "height": 24,
                "lineHeight": 24,
                "marginVertical": 4,
                "textAlignVertical": "center",
              },
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
                "marginLeft": 4,
                "marginRight": 8,
              },
            ],
          ]
        }
        theme={
          Object {
            "colors": Object {
              "accent": "#03dac4",
              "backdrop": "rgba(0, 0, 0, 0.5)",
              "background": "#f6f6f6",
              "disabled": "rgba(0, 0, 0, 0.26)",
              "error": "#B00020",
              "placeholder": "rgba(0, 0, 0, 0.54)",
              "primary": "#6200ee",
              "surface": "#ffffff",
              "text": "#000000",
            },
            "dark": false,
            "fonts": Object {
              "light": "HelveticaNeue-Light",
              "medium": "HelveticaNeue-Medium",
              "regular": "Helvetica Neue",
              "thin": "HelveticaNeue-Thin",
            },
            "roundness": 4,
          }
        }
      >
        Example Chip
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders outlined disabled chip 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "rgba(0, 0, 0, 0.29)",
      "borderRadius": 16,
      "borderStyle": "solid",
      "borderWidth": 0.5,
      "elevation": 0,
    }
  }
>
  <View
    accessibilityComponentType="button"
    accessibilityTraits={
      Array [
        "button",
        "disabled",
      ]
    }
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderRadius": 16,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "paddingHorizontal": 4,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "Helvetica Neue",
              "textAlign": "left",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "height": 24,
                "lineHeight": 24,
                "marginVertical": 4,
                "textAlignVertical": "center",
              },
              Object {
                "color": "rgba(0, 0, 0, 0.26)",
                "marginLeft": 8,
                "marginRight": 8,
              },
            ],
          ]
        }
        theme={
          Object {
            "colors": Object {
              "accent": "#03dac4",
              "backdrop": "rgba(0, 0, 0, 0.5)",
              "background": "#f6f6f6",
              "disabled": "rgba(0, 0, 0, 0.26)",
              "error": "#B00020",
              "placeholder": "rgba(0, 0, 0, 0.54)",
              "primary": "#6200ee",
              "surface": "#ffffff",
              "text": "#000000",
            },
            "dark": false,
            "fonts": Object {
              "light": "HelveticaNeue-Light",
              "medium": "HelveticaNeue-Medium",
              "regular": "Helvetica Neue",
              "thin": "HelveticaNeue-Thin",
            },
            "roundness": 4,
          }
        }
      >
        Example Chip
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders selected chip 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(0, 0, 0, 0.24)",
      "borderColor": "#ebebeb",
      "borderRadius": 16,
      "borderStyle": "solid",
      "borderWidth": 0.5,
      "elevation": 0,
    }
  }
>
  <View
    accessibilityComponentType="button"
    accessibilityTraits={
      Array [
        "button",
        "selected",
      ]
    }
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderRadius": 16,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "paddingHorizontal": 4,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "padding": 4,
            },
            null,
          ]
        }
      >
        <Text
          accessibilityElementsHidden={true}
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          style={
            Array [
              Object {
                "color": "rgba(0, 0, 0, 0.54)",
                "fontSize": 18,
              },
              Array [
                Object {
                  "transform": Array [
                    Object {
                      "scaleX": 1,
                    },
                  ],
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "Helvetica Neue",
              "textAlign": "left",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "height": 24,
                "lineHeight": 24,
                "marginVertical": 4,
                "textAlignVertical": "center",
              },
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
                "marginLeft": 4,
                "marginRight": 8,
              },
            ],
          ]
        }
        theme={
          Object {
            "colors": Object {
              "accent": "#03dac4",
              "backdrop": "rgba(0, 0, 0, 0.5)",
              "background": "#f6f6f6",
              "disabled": "rgba(0, 0, 0, 0.26)",
              "error": "#B00020",
              "placeholder": "rgba(0, 0, 0, 0.54)",
              "primary": "#6200ee",
              "surface": "#ffffff",
              "text": "#000000",
            },
            "dark": false,
            "fonts": Object {
              "light": "HelveticaNeue-Light",
              "medium": "HelveticaNeue-Medium",
              "regular": "Helvetica Neue",
              "thin": "HelveticaNeue-Thin",
            },
            "roundness": 4,
          }
        }
      >
        Example Chip
      </Text>
    </View>
  </View>
</View>
`;
